src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -33,56 +34,112 @@ public PropertySubpatternCompletionProvider()
+                    displayText: member.Name.EscapeIdentifier(),
                     displayTextSuffix: ColonString,
                     insertionText: null,
-                    symbols: ImmutableArray.Create(untestedMember),
-                    contextPosition: token.GetLocation().SourceSpan.Start,
+                    symbols: ImmutableArray.Create(member),
+                    contextPosition: context.Position,
                     rules: s_rules));
             }
+
+            return;
+
+            // We have to figure out the type of the extended property ourselves, because
+            // the semantic model could not provide the answer we want in incomplete syntax:
+            // `c is { X. }`
+            static ITypeSymbol GetMemberAccessType(ITypeSymbol type, ExpressionSyntax expression, Document document, SemanticModel semanticModel, int position)
+            {
+                if (expression is null)
+                {
+                    return type;
+                }
+                else if (expression is MemberAccessExpressionSyntax memberAccess)
+                {
+                    type = GetMemberAccessType(type, memberAccess.Expression, document, semanticModel, position);
+                    return GetMemberType(type, name: memberAccess.Name.Identifier.ValueText, document, semanticModel, position);
+                }
+                else if (expression is IdentifierNameSyntax identifier)
+                {
+                    return GetMemberType(type, name: identifier.Identifier.ValueText, document, semanticModel, position);
+                }
+
+                throw ExceptionUtilities.Unreachable;
+            }
+
+            static ITypeSymbol GetMemberType(ITypeSymbol type, string name, Document document, SemanticModel semanticModel, int position)
+            {
+                var members = GetCandidatePropertiesAndFields(document, semanticModel, position, type);
+                var matches = members.WhereAsArray(m => m.Name == name);
+                if (matches.Length != 1)
+                {
+                    return null;
+                }
+
+                return matches[0] switch
+                {
+                    IPropertySymbol property => property.Type,
+                    IFieldSymbol field => field.Type,
+                    _ => throw ExceptionUtilities.Unreachable,
+                };
+            }
+
+            static ImmutableArray<ISymbol> GetCandidatePropertiesAndFields(Document document, SemanticModel semanticModel, int position, ITypeSymbol type)
+            {
+                var members = semanticModel.LookupSymbols(position, type);
+                return members.WhereAsArray(m => m.CanBeReferencedByName &&
+                    IsFieldOrReadableProperty(m) &&
+                    !m.IsImplicitlyDeclared &&
+                    !m.IsStatic &&
+                    m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
```

Comment:
i dont' think IsEditorBrowsable is right in this method (it should be at the top level).  Wit hthis approach, if you had:

```x.y.z.``` you wouldn't show anything if `y` was EB(never).  Instead, we want to still lookup the members properly, but just not show the EB properties/fields at the *end* when determining the final list of members to show.

Reply:
Good catch.
Fixed but FYI I wasn't able to test (attribute is ignored in source).

Reply:
in general we just do a cross language test.  write the type in VB, consume in C#.    But i'm ok if you dont' want to write such a test here.

Reply:
Thanks, that worked (added test that fails before fix but passes now) :-)